src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -107,19 +82,30 @@ void snc_quotas_probe::setup_metrics() {
           sm::description(
             "Number of times throughput quota balancer has been run")));
     }
-    add_ingress_egress_gauges(
-      metric_defs,
-      "quota_effective",
-      [this] { return _qm.get_quota(); },
-      sm::description("Currently effective quota, in bytes/s"),
-      aggregate_labels);
-    add_ingress_egress_gauges(
-      metric_defs,
-      "throughput",
-      [this] { return _qm.get_throughput(); },
-      sm::description("Throughput measurement at the time of last "
-                      "balancer run, in bytes/s"),
-      aggregate_labels);
+    {
+        static const char* name = "quota_effective";
+        static const auto desc = sm::description(
+          "Currently effective quota, in bytes/s");
+        metric_defs.emplace_back(
+          sm::make_counter(
+            name, [this] { return _qm.get_quota().in; }, desc, {label_ingress})
+            .aggregate(aggregate_labels));
+        metric_defs.emplace_back(
+          sm::make_counter(
+            name, [this] { return _qm.get_quota().eg; }, desc, {label_egress})
+            .aggregate(aggregate_labels));
+    }
+    {
+        static const char* name = "traffic";
+        static auto desc = sm::description(
+          "Amount of Kafka traffic processed, in bytes");
```

Comment:
Why a counter over a gauge? And don't we already have a metric that represents the amount of kafka traffic processed in bytes? If I recall correctly its the built-in metrics within the rpc layer.

Reply:
> Why a counter over a gauge

Because it's a monotonically increasing value. https://prometheus.io/docs/concepts/metric_types/#counter

Reply:
> And don't we already have a metric that represents the amount of kafka traffic processed in bytes? If I recall correctly its the built-in metrics within the rpc layer.

I have fund the metric called `vectorized_kafka_rpc_received_bytes`. The difference is that this metric is collected as soon as we have received the request from rpc, and therefore it's not suitable for accounting for throttled TP.

The other issue is that probably this new metric should be collected in the context of `kafka::server::probe` rather than `snc_quota_manager::probe` and named in a way to align with these ones, like `vectorized_kafka_rpc_consumed_bytes`. I take this as AI, feedback on this is welcome.